e2e/testcases/override_reconcile_timeout_test.go

```diff
@@ -68,88 +78,55 @@ func TestOverrideReconcileTimeout(t *testing.T) {
 	nt.RootRepos[configsync.RootSyncName].Add("acme/ns-1.yaml", fake.NamespaceObject(namespaceName))
 	nt.RootRepos[configsync.RootSyncName].CommitAndPush(fmt.Sprintf("Add namespace/%s & pod/%s (never ready)", namespaceName, pod1Name))
 	nt.WaitForRepoSyncs()
+	expectActuationStatus := "Succeeded"
+	expectReconcileStatus := "Timeout"
+	nomostest.WatchForObject(nt, kinds.ResourceGroup(), "root-sync", "config-management-system",
+		[]nomostest.Predicate{
+			resourceStatusEquals(idToVerify, expectActuationStatus, expectReconcileStatus),
+		})
 
-	u := &unstructured.Unstructured{}
-	u.SetGroupVersionKind(schema.GroupVersionKind{
-		Group:   "kpt.dev",
-		Kind:    "ResourceGroup",
-		Version: "v1alpha1",
-	})
-
-	err := nt.Client.Get(nt.Context, client.ObjectKey{
-		Namespace: "config-management-system",
-		Name:      "root-sync",
-	}, u)
-	if err != nil {
-		nt.T.Fatal(err)
-	} else {
-		expectActuationStatus := "Succeeded"
-		expectReconcileStatus := "Timeout"
-		err = verifyObjectStatus(u, idToVerify, expectActuationStatus, expectReconcileStatus)
-		if err != nil {
-			nt.T.Fatal(err)
-		} else {
-			nt.T.Logf("object %q has actuation and reconcile status as expected", namespaceName)
-		}
-	}
+	// Override reconcileTimeout to 5m, namespace actuation should succeed, namespace reconcile should succeed.
+	nt.MustMergePatch(rootSync, `{"spec": {"override": {"reconcileTimeout": "5m"}}}`)
```

Comment:
This is likely going to be flakey.

Unfortunately, we don't actually have any signal that the spec change has taken affect. This will be fixed when we add `syncedGeneration` or similar to the RSync status.

In the mean time, you might want to add a WatchForObject with a predicate that ensures the `observedGeneration` has been updated to match the new value, before calling WaitForRepoSyncs. That way we know the reconciler-manager at least saw the new reconcileTimeout.

To do so, you'll also need to Patch the object directly, instead of using MustMergePatch. That way you can read the new generation out of the object from the result of the Patch call. Maybe make a local function to make this readable.

Reply:
Once we change the field spec.override.reconcileTimeout in Rsync, a new reconcile process start, the reconciler pod will be recreated and the Rsync status will change to be:       
      message: 'Available: 0/1'
      reason: Deployment
      status: "True"
      type: Reconciling
And the WaitForRepoSyncs() will detect the the deployment reconciling status is true, it will wait until the deployment is ready again to return. So this Patch should not be flakey.

Reply:
You're assuming that the Reconciling condition will be set to True by the reconciler-manager before  WaitForRepoSyncs does its first GET on the RSync. But it's possible the reconciler-manager is slow or delayed, which could cause the test GEt to happen before the reconicler-manager has updated the reconciler and RSync. That's a race condition.

The way to prove that the reconciler-manager has seen the desired RSync spec and that it's updated the Reconciling condition is to check that the RSync's status.observedGeneration has been updated to be greater than or equal to the desired generation (the new generation returned from the RSync UPDATE call).